apps/microtvm/arduino/template_project/microtvm_api_server.py

```diff
@@ -341,16 +376,49 @@ def generate_project(self, model_library_format_path, standalone_crt_dir, projec
         shutil.copy2(model_library_format_path, project_dir / MODEL_LIBRARY_FORMAT_RELPATH)
 
         # For AOT, template model.h with metadata to minimize space usage
-        if options["project_type"] == "example_project":
+        if project_type == "example_project":
             self._template_model_header(source_dir, metadata)
 
         self._change_cpp_file_extensions(source_dir)
 
         # Recursively change includes
         self._convert_includes(project_dir, source_dir)
 
-    def _get_arduino_cli_cmd(self, options: dict):
-        arduino_cli_cmd = options.get("arduino_cli_cmd", ARDUINO_CLI_CMD)
+        # create include directory
+        (project_dir / "include").mkdir()
+        build_extra_flags = "-I./include"
+
+        if self._cmsis_required(project_dir):
+            build_extra_flags += f" -I./include/cmsis"
+            self._copy_cmsis(project_dir, cmsis_path)
+
+        # Populate Makefile
+        with open(project_dir / MAKEFILE_FILENAME, "w") as makefile_f:
```

Comment:
Can we make the code that populates the Makefile its own function? That's what we do for `_change_cpp_file_extensions`, `_convert_includes`, `_copy_cmsis`, etc.